azkaban-common/src/main/java/azkaban/executor/ExecutableFlowPriorityComparator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2014 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor;
+
+import java.util.Comparator;
+
+import org.apache.log4j.Logger;
+
+import azkaban.utils.Pair;
+
+public final class ExecutableFlowPriorityComparator implements
+  Comparator<Pair<ExecutionReference, ExecutableFlow>> {
+  private static Logger logger = Logger
+    .getLogger(ExecutableFlowPriorityComparator.class);
+
+  @Override
+  public int compare(Pair<ExecutionReference, ExecutableFlow> pair1,
```

Comment:
Please add javadoc for the comparison logic - descending order by priority,  ascending order by update time and ascending order by execution id